menas/ui/components/dataset/datasetMain.controller.js

```diff
@@ -238,10 +238,11 @@ sap.ui.controller("components.dataset.datasetMain", {
         onClose: function (oResponse) {
           if (oResponse === sap.m.MessageBox.Action.YES) {
             let toks = sBindPath.split("/");
-            let index = toks[toks.length - 1];
-            let currDataset = this._model.getProperty("/currentDataset");
-            let conformance = currDataset["conformance"].filter((el, ind) => ind !== parseInt(index));
-            // RuleService.editConformanceRules(currDataset.name, currDataset.version, conformance);
+            let ruleIndex = parseInt(toks[toks.length - 1]);
+            let currentDataset = this._model.getProperty("/currentDataset");
+            let newDataset = RuleService.removeRule(currentDataset, ruleIndex);
+
+            if(newDataset) DatasetService.editDataset(newDataset)
```

Comment:
Why the `if` clause?
If it's to remain the please fix to adhere to coding standards:
```
if (newDataset) {
  DatasetService.editDataset(newDataset);
}
```

Reply:
In next iteration will removeRule return null because of validation.
Style fixed on if.

Reply:
This future addressing development can be confusing. Not mentioning when the suggested happens it still needs to be checked if the calls handle the changed result.
Thanks for the style fix.